kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java

```diff
@@ -43,7 +43,7 @@ public final class JobItemInventoryTasksProgress {
      * @return inventory position
      */
     public Map<String, IngestPosition> getInventoryPosition(final String tableName) {
-        Pattern pattern = Pattern.compile(String.format("%s(#\\d+)?", tableName));
+        Pattern pattern = Pattern.compile(String.format("\\.%s#(\\d+)?", tableName));
```

Comment:
Is `#` always existing?

From old code:
```
    private String generateTaskId(final InventoryDumperConfiguration inventoryDumperConfig) {
        String result = String.format("%s.%s", inventoryDumperConfig.getDataSourceName(), inventoryDumperConfig.getActualTableName());
        return null == inventoryDumperConfig.getShardingItem() ? result : result + "#" + inventoryDumperConfig.getShardingItem();
    }
```

Reply:
If using `InventoryTaskSplitter`, the shardingItem wouldn't be null, scenario not found where shardingItem is null

Reply:
OK